src/upgrade/UpgradeToMaltainvestCard.js

```diff
@@ -46,119 +47,66 @@ export default class UpgradeToMaltainvestCard extends PureComponent {
+        last_name: props.last_name,
+        date_of_birth: moment.unix(this.props.date_of_birth).format('YYYY-MM-DD') || '1980-01-01',
+        place_of_birth: props.place_of_birth,
+        address_line_1: props.address_line_1,
+        address_line_2: props.address_line_2,
+        address_city: props.address_city,
+        address_state: props.address_state,
+        address_postcode: props.address_postcode,
+        secret_question: props.secret_question,
+        secret_answer: props.secret_answer,
+        phone: props.phone,
+      }
     };
-  }
-
-  componentWillMount() {
-    this.setState({ date_of_birth: moment.unix(this.props.date_of_birth).format('YYYY-MM-DD') || '1980-01-01' });
-    this.validateForm();
-  }
 
-  validateForm = () => {
     this.constraints = getConstraints(this.props);
-    this.setState({
-      errors: validate(this.state, this.constraints, { format: 'grouped', fullMessages: false, cleanAttributes: false }) || {},
-    });
+    this.validationMan = new ValidationManager(this.constraints);
   }
 
   onEntryChange = (e: SyntheticEvent) => {
-    this.setState({
-      [e.target.id]: e.target.value,
-      touched: { ...this.state.touched, [e.target.id]: true },
-      hasError: false,
-    }, () => {
-      this.validateForm();
-    });
+    const s = this.validationMan.validateFieldAndGetNewState(e, this.state.formData);
+		this.setState({ ...s, hasError: false });
   }
 
   onCountryChange = (e: SyntheticEvent) => {
-    this.setState({ [e.target.id]: e.target.value, hasError: false, });
+    this.onEntryChange(e);
     api.getStatesForCountry(e.target.value).then(response =>
       this.setState({ statesList: response.states_list })
     );
-    this.validateForm();
   }
 
-  onTaxResidenceChange = (val) => this.setState({ tax_residence: val,
-    /*eslint-disable */
-    touched: { ...this.state.touched, 'tax_residence': true },
-    /*eslint-enable */
-    hasError: false,
-  }, () => {
-    this.validateForm();
-  });
+  onTaxResidenceChange = (val) => {
+		const s = this.validationMan.validateAndGetNewState('tax_residence', val, this.state.formData);
+		this.setState({ ...s, hasError: false });
+	}
 
   onTermsAndConditionsChanged = (e: SyntheticEvent) => {
-    const accept_risk = e.target.checked ? '1' : '0';
-    this.setState({ accept_risk, touched: { ...this.state.touched, [e.target.id]: true }, });
+    // accept_risk takes a boolean value of either 1 or 0.
+    const val = e.target.checked ? 1 : 0;
+    const s = this.validationMan.validateAndGetNewState('accept_risk', val, this.state.formData);
+		this.setState({ ...s, hasError: false });
   }
 
   onPEPDeclarationChanged = (e: SyntheticEvent) => {
-    const PEPDeclaration = e.target.checked ? '1' : '0';
-    this.setState({ PEPDeclaration, touched: { ...this.state.touched, [e.target.id]: true }, });
+    const val = e.target.checked ? 1 : 0;
+    const s = this.validationMan.validateAndGetNewState('PEPDeclaration', val, this.state.formData);
```

Comment:
@bruce I think here PEPDeclaration is in formData which is going to be in request, right?
 but it's just a checkbox which should be checked to allow the users to upgrade their account, it's not included in the request, so when we include it in request user will get an error in response that it's not defined in the schema.
could you please check this again?

Reply:
I'm currently getting access to a QA box to test this, but in the mean time this is a simple fix. I'll put up a PR for this by today.

Reply:
This should be fixed in https://github.com/binary-com/binary-next-gen/pull/716